model/search.js

```diff
@@ -0,0 +1,214 @@
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+      },
+    }));
+  const fuseIndex = fuseResp && Fuse.parseIndex(await fuseResp.json());
+
+  const allItems = getItemAll(garden);
+  if (allItems.length > 0) {
+    let itemsToUpdate = [];
+    if (fuseLastModified) {
+      for (const item of allItems) {
+        if (getDatetime(item, DCTERMS.modified) > fuseLastModified) {
+          itemsToUpdate = [...itemsToUpdate, item];
+        }
+      }
+    } else {
+      itemsToUpdate = allItems;
+    }
+
+    if (itemsToUpdate.length > 0) {
+      const entriesToUpdate = await fullTextFuseEntriesFromGardenItems(
+        itemsToUpdate,
+        gardenUrl,
+        {
+          fetch,
+        }
+      );
+      console.log(`Entries to update: ${entriesToUpdate.length}`);
+      const allEntries = fuseEntriesFromGardenItems(allItems, gardenUrl);
+      console.log(`Total entries: ${allEntries.length}`);
+      const options = defaultOptions(FullTextFuseKeys);
+      let fuse;
+      if (fuseIndex) {
+        fuse = new Fuse(allEntries, options, fuseIndex);
+        const toRemove = new Set();
+        for (const item of itemsToUpdate) {
+          toRemove.add(getUUID(item));
+        }
+        fuse.remove((doc) => {
+          return toRemove.has(doc.uuid);
+        });
+        for (const entry of entriesToUpdate) {
+          fuse.add(entry);
+        }
+      } else {
+        fuse = new Fuse(entriesToUpdate, options);
+      }
+
+      const updatedIndex = fuse.getIndex();
+      console.log('Updating saved json index');
+      const buf = Buffer.from(JSON.stringify(updatedIndex.toJSON()));
+      await overwriteFile(fuseIndexUrl, buf, {
+        fetch,
+        contentType: 'application/json',
+      });
+
+      const updatedFuseIndexInfo = setDatetime(
+        fuseIndexInfo || createThing({ url: fuseIndexUrl }),
+        DCTERMS.modified,
+        new Date()
+      );
+      await saveSolidDatasetAt(
+        gardenUrl,
+        setThing(garden, updatedFuseIndexInfo),
+        {
+          fetch,
+        }
+      );
+    } else {
+      console.log(
+        `Index last modified at ${fuseLastModified}.  Nothing to update.`
+      );
+    }
+  }
+}
+
+export async function setupGardenSearchIndex(
+  gardenUrl,
+  fuseIndexUrl,
+  { fetch }
+) {
+  console.log(
+    `Setting up garden search index for ${gardenUrl} at ${fuseIndexUrl}`
+  );
+  const garden = await getSolidDataset(gardenUrl, { fetch });
+  return setupOrUpdateGardenSearchIndex(garden, fuseIndexUrl, { fetch });
+}
+
+export async function updateGardenSearchIndex(gardenUrl, { fetch }) {
+  console.log(`Updating garden search index for ${gardenUrl}`);
+  const garden = await getSolidDataset(gardenUrl, { fetch });
+  const gardenSettings = getThing(garden, gardenUrl);
+  const fuseIndexUrl = getUrl(gardenSettings, MY.Garden.hasFuseIndex);
+  if (fuseIndexUrl) {
+    console.log(`Found garden search index at ${fuseIndexUrl}`);
```

Comment:
here and above too? or are these console.logs intended to be left in for debugging for now?

Reply:
All the ones in model/search are backend logs.